operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -563,7 +563,7 @@ private void updateResourceState(Reconciliation reconciliation, boolean ready, T
             LOGGER.debugCr(reconciliation, "Removed metric " + METRICS_PREFIX + "resource.state{}", key);
         }
 
-        if (cr != null) {
+        if (cr != null && Util.matchesSelector(selector(), cr)) {
```

Comment:
Why is this ever called for resources not matching the selector?

Reply:
I guess because we need to delete metrics for the resources not matching the selector.

Reply:
But this does not even have an else. So how does it delete the metric?

Reply:
In the code above (line 562). It looks for the metric, removes it, and only if the CR is nonNull (and matches the selector), the metric is put back - updated.

Reply:
Ok. 

But I'm still not sure this fixes the issue. `updateResourceState` should be called at the end of the reconciliation. So the  change you did should catch situations when the CR moved out of the selector during the reconciliation it self. However, that:
* Should not be done AFAIK if the handover happens properly and the reconciliation is paused, or? It can be done only when it is done without pausing in which case all kind of bad things can happen
* If that is really the case where it happened in #5293, the metric should be there only for a short time and be removed later anyway.

Isn't it the case that we need to delete the metrics when the resource is deleted?